include/ignition/gazebo/comms/MsgManager.hh

```diff
@@ -0,0 +1,162 @@
+  /// posible to bind to this address using a different model.
+  /// \param[in] _address The subscriber address.
+  /// \param[in] _modelName The model name.
+  /// \param[in] _topic The subscriber topic.
+  /// \return True if the subscriber was successfully added or false otherwise.
+  public: bool AddSubscriber(const std::string &_address,
+                             const std::string &_modelName,
+                             const std::string &_topic);
+
+  /// \brief Add a new message to the inbound queue.
+  /// \param[in] _address The destination address.
+  /// \param[in] _msg The message.
+  public: void AddInbound(const std::string &_address,
+                          const msgs::DataframeSharedPtr &_msg);
+
+  /// \brief Add a new message to the outbound queue.
+  /// \param[in] _address The sender address.
+  /// \param[in] _msg The message.
+  public: void AddOutbound(const std::string &_address,
+                           const msgs::DataframeSharedPtr &_msg);
+
+  /// \brief Remove an existing subscriber.
+  /// \param[in] _address The subscriber address.
+  /// \param[in] _topic The Subscriber topic.
+  /// \return True if the subscriber was removed or false otherwise.
+  public: bool RemoveSubscriber(const std::string &_address,
+                                const std::string &_topic);
+
+  /// \brief Remove a message from the inbound queue.
+  /// \param[in] _address The destination address.
+  /// \param[in] _Msg Message pointer to remove.
+  /// \return True if the message was removed or false otherwise.
+  public: bool RemoveInbound(const std::string &_address,
+                             const msgs::DataframeSharedPtr &_msg);
+
+  /// \brief Remove a message from the outbound queue.
+  /// \param[in] _address The sender address.
+  /// \param[in] _msg Message pointer to remove.
+  /// \return True if the message was removed or false otherwise.
+  public: bool RemoveOutbound(const std::string &_address,
+                              const msgs::DataframeSharedPtr &_msg);
+
+  /// \brief This function delivers all the messages in the inbound queue to
+  /// the appropriate subscribers. This function also clears the inbound queue.
+  public: void DeliverMsgs();
+
+  /// \brief Get a mutable reference to the data containing subscriptions and
+  /// data queues.
+  /// \return A mutable reference to the data.
+  public: const Registry &DataConst();
+
+  /// \brief Get a mutable reference to the data containing subscriptions and
+  /// data queues.
+  /// \return A mutable reference to the data.
+  public: Registry &Data();
+
+  /// \brief Get a copy of the data structure containing subscriptions and data
+  /// queues.
+  /// \return A copy of the data.
+  public: Registry Copy();
+
+  /// \brief Set the data structure containing subscriptions and data queues.
+  /// \param[in] _newContent New content to be set.
+  public: void Set(Registry &_newContent);
```

Comment:
checking that the arg is meant to be passed as ref and not const ref?

Reply:
Changed in [5472d97](https://github.com/ignitionrobotics/ign-gazebo/pull/1416/commits/5472d97e1993d96147318f3a6c834f1026067232).